### DISCUSSION TITLE
Is there a faster way to swap GIFs?

### DISCUSSION THREAD

**Discussion by LambMediaGroup** [thelastoutpostworkshop/animated_gif_memory#3] [Q&A]
I have about 10 gifs that I've got configured for this, is there a faster way to switch between them?

**Comment by thelastoutpostworkshop** [comment]
I am not sure what you mean exactly by faster way to switch between them, if your start playing a GIF and you have a switch to stop it and start another, it will played almost immediately since it is in memory.  But 10 GIFs, they have to be small in order to fit in the microcontroller memory.

**Reply by LambMediaGroup** [reply]
Currently, if I want to change the GIF that's being displayed I have to plug the board into my PC, open the arduino software, comment out the line that's currently active and uncomment one of the other GIFs and then complile/upload. 

I was wondering if there is some way that I could update the GIF without doing all that. Maybe if I hosted all the gifs on a Google Drive or Dropbox and then just called one of the GIF files **current.h** and the system could periodically download that file and replace the one in memory with the one downloaded, that way when I change that file it'll automatically update on the screen. Can that be done or is there any other way to replace the current GIF with another?

**Comment by thelastoutpostworkshop** [comment]
You will have to to do it over Wifi, it's feasible but it's more advanced considering that Onedrive and Dropbox are secured storage, You have to know how the wifi security works in the wifi library of the microcontroller.
You could also use a storage on your local wifi on your PC for example avoiding the security aspect.
The easiest way is to store them on an SD card like I show in the tutorial.

**Comment by LambMediaGroup** [comment]
Sorry, this is my first project so I don't have much knowledge of coding. I have managed to get it onto my wifi though. If the files were available on an unsecured website would that make it easier to achieve?

If not maybe I'll just try the SD card method.

**Comment by thelastoutpostworkshop** [comment]
Yes but you have to host locally the unsecure website on a pc or any computer on your local network
